src/modules/Elsa.EntityFrameworkCore/Modules/Runtime/ActivityExecutionLogStore.cs

```diff
@@ -58,42 +73,56 @@ public EFCoreActivityExecutionStore(EntityStore<RuntimeElsaDbContext, ActivityEx
 
     private async ValueTask OnSaveAsync(RuntimeElsaDbContext dbContext, ActivityExecutionRecord entity, CancellationToken cancellationToken)
     {
-        dbContext.Entry(entity).Property("SerializedActivityState").CurrentValue = entity.ActivityState != null ? await _safeSerializer.SerializeAsync(entity.ActivityState, cancellationToken) : default;
+        var compressionAlgorithm = _options.Value.CompressionAlgorithm ?? nameof(None);
+        var serializedActivityState = entity.ActivityState != null ? await _safeSerializer.SerializeAsync(entity.ActivityState, cancellationToken) : default;
+        var compressedSerializedActivityState = serializedActivityState != null ? await _compressionStrategyResolver.Resolve(compressionAlgorithm).CompressAsync(serializedActivityState, cancellationToken) : default;
```

Comment:
We do not have to resolve the strategy for each call unless we want to be able to change options while running.

Reply:
True, we could store it as a field on this class, or introduce a "compression provider" type that resolves the active compression strategy once.

Reply:
On second thought, each record may use a different algorithm, since the setting may change over time. Therefore it's important to resolve the correct algo per record. A small optimization can be done however by using a dictionary for quicker lookups.